BUG/API: Set geometry api where `col` is named / with `drop=True` (#3237)
geopandas/geopandas: CHANGELOG.md

Backwards incompatible API changes:
- The deprecated default value of GeoDataFrame/ GeoSeries `explode(.., index_parts=True)` is now
  set to false for consistency with pandas (#3174)

> - The behaviour of `set_geometry` has been updated when `col` is a (Geo)Series with `ser.name` 
>   not None. The new active geometry column name in this case will be `ser.name`, rather than

Reviewer: ```suggestion
- The behaviour of `set_geometry` has been changed when passed a (Geo)Series `ser` with a name. 
  The new active geometry column name in this case will be `ser.name`, if not None, rather than
```

Backwards incompatible API changes:
- The deprecated default value of GeoDataFrame/ GeoSeries `explode(.., index_parts=True)` is now
  set to false for consistency with pandas (#3174)
- The behaviour of `set_geometry` has been updated (#3237):

>    - When `col` is a str and `drop=True`, the new active geometry column name will be `col`,

Reviewer: ```suggestion
   - When `col` is an existing column name and `drop=True`, the new active geometry column name will be `col`,
```

     rather than the previous active geometry column name
   - When `col` is a (Geo)Series with `ser.name` not None, the new active geometry column name
     will be `ser.name`, rather than the previous active geometry column name. This means that
     `drop` now also functions when `col` is a (Geo)Series, and can be used to remove the 

>      previous active geometry column name when it differs from `ser.name`.

Author: I think we should also throw if drop is given when `col` is an arraylike, but haven't done that.

  for the default pyogrio engine. Currently those are translated to the ``columns`` keyword
  for backwards compatibility, but you should directly use the ``columns`` keyword instead
  to select which columns to read (#3133).
- The `drop` keyword in `set_geometry` has been deprecated, and in future the `drop=True` 

>   behaviour will be removed. To prepare for this change, you should remove any explict 

Reviewer: ```suggestion
  behaviour will be removed. To prepare for this change, you should remove any explicit 
```

  `drop=False` calls in your code (the default behaviour already is the same as `drop=False`).
  To replicate the previous `drop=True` behaviour you should replace 
  `gdf.set_geometry(new_geo_col, drop=True)` with

> ```python

Reviewer: I assume this will need to be intended the same as the list content
